.github/workflows/plugins_react_tests.yml

```
@@ -13,48 +13,60 @@ permissions:
   contents: read
 
 jobs:
+  setup_matrix:
+    name: Setup matrix
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.build_matrix.outputs.matrix }}
+    steps:
+      - name: Build test matrix
+        id: build_matrix
+        uses: ekohl/gha-matrix-builder@v0
+
   test:
+    name: ${{ matrix.plugin }} with Ruby ${{ matrix.ruby }} and Node ${{ matrix.node }}
     runs-on: ubuntu-latest
+    needs: setup_matrix
     timeout-minutes: 20
     strategy:
       fail-fast: false
       matrix:
-        include:
-          - repo: katello
-            org: Katello
-          - repo: foreman-tasks
-            org: theforeman
+        ruby: ${{ fromJson(needs.setup_matrix.outputs.matrix).ruby }}
+        node: ${{ fromJson(needs.setup_matrix.outputs.matrix).node }}
+        plugin:
+          - theforeman/foreman-tasks
+          - Katello/katello
     steps:
       - name: Use Node.js
         uses: actions/setup-node@v3
```

Didn't we update this to v4?
```suggestion
        uses: actions/setup-node@v4
```